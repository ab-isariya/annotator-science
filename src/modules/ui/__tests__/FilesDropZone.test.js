import renderer from 'react-test-renderer';

import FilesDropZone from '../FilesDropzone';

describe('<FilesDropZone/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer
      .create(<FilesDropZone onFileDrop={jest.fn()} />)
      .toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with a custom classname without crashing', () => {
    const comp = renderer
      .create(<FilesDropZone onFileDrop={jest.fn()} className="testClass" />)
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/testClass/i);
  });
});
