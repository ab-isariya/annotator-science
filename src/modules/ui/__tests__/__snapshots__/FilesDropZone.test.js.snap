// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilesDropZone/> Renders with a custom classname without crashing 1`] = `
<section
  className="dropzone h-full min-h-full w-full bg-grey-50 border-blue-500 border-2 border-dashed rounded-md flex items-center justify-center testClass"
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <input
    accept="application/pdf, application/vnd.openxmlformats-officedocument.wordprocessingml.document, application/msword"
    autoComplete="off"
    multiple={true}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />
  <div
    className="min-h-full flex flex-col justify-center items-center font-inter"
  >
    <div
      className="mb-4"
    >
      <div
        className=" relative"
      >
        <svg
          style={
            Object {
              "transform": "scale(1.3)",
            }
          }
        >
          IconBorder-46.svg
        </svg>
        <div
          className="absolute top-2/4 left-2/4"
          style={
            Object {
              "transform": "translate(-50%, -50%)",
            }
          }
        >
          <svg
            style={
              Object {
                "filter": " invert(39%) sepia(17%) saturate(3351%) hue-rotate(200deg) brightness(95%) contrast(89%)",
              }
            }
          >
            DragAndDrop.svg
          </svg>
        </div>
      </div>
    </div>
    <div
      className="font-light text-2xl leading-9"
    >
      Drag and drop file or click to upload
    </div>
    <div
      className="mt-1 font-normal text-base leading-6"
    >
      Only .PDF and .DOCX files will be accepted
    </div>
  </div>
</section>
`;

exports[`<FilesDropZone/> Renders without crashing 1`] = `
<section
  className="dropzone h-full min-h-full w-full bg-grey-50 border-blue-500 border-2 border-dashed rounded-md flex items-center justify-center "
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <input
    accept="application/pdf, application/vnd.openxmlformats-officedocument.wordprocessingml.document, application/msword"
    autoComplete="off"
    multiple={true}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />
  <div
    className="min-h-full flex flex-col justify-center items-center font-inter"
  >
    <div
      className="mb-4"
    >
      <div
        className=" relative"
      >
        <svg
          style={
            Object {
              "transform": "scale(1.3)",
            }
          }
        >
          IconBorder-46.svg
        </svg>
        <div
          className="absolute top-2/4 left-2/4"
          style={
            Object {
              "transform": "translate(-50%, -50%)",
            }
          }
        >
          <svg
            style={
              Object {
                "filter": " invert(39%) sepia(17%) saturate(3351%) hue-rotate(200deg) brightness(95%) contrast(89%)",
              }
            }
          >
            DragAndDrop.svg
          </svg>
        </div>
      </div>
    </div>
    <div
      className="font-light text-2xl leading-9"
    >
      Drag and drop file or click to upload
    </div>
    <div
      className="mt-1 font-normal text-base leading-6"
    >
      Only .PDF and .DOCX files will be accepted
    </div>
  </div>
</section>
`;
